### DISCUSSION TITLE
React vs Preact for Components

### DISCUSSION THREAD

**Discussion by tanishqmanuja** [satnaing/astro-paper#124] [Polls]
### React:
![react](https://github.com/satnaing/astro-paper/assets/40914272/cd7907fa-9eb3-41d4-b2ba-ea25325ad5fb)

### Preact:
![preact](https://github.com/satnaing/astro-paper/assets/40914272/6a0ffa22-b95f-40f1-b5c5-76cca3ffb861)

### Note
- Network traffic is measured using a full page reload with cache disabled
- Ref commit for diff https://github.com/tanishqmanuja/blog/commit/cf0b21d2b154040833c328d507f1829d11e11cc4

**Comment by satnaing** [comment]
How about the UX, is that noticeable?
And what about the bundled size?

I've heard of `Preact` for some time, but haven't tried it yet TBH.

**Reply by tanishqmanuja** [reply]
Yeah bundle size is less, same can be seen if you sum all the js files in the network tab. At a glance you will notice that in the case of react, the client bundle is 45kb!

**Reply by tanishqmanuja** [reply]
As for the actual user experience you cant tell if the site is using react or preact, unless you open dev tools. But preact loads faster :)
